test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1285,46 +1285,50 @@ def test_cgroup_periodic_update_check_values(self):
-        if self.paths['memory']:
-            lines = self.moms_list[0].log_match(
-                '%s;update_job_usage: Memory usage: mem=' % jid,
-                allmatch=True, max_attempts=5, starttime=begin)
-            usage = 0
-            for line in lines:
-                match = re.search(r'mem=(\d+)kb', line[1])
-                if not match:
-                    continue
-                usage = int(match.groups()[0])
-                if usage > 400000:
-                    break
-            self.assertGreater(usage, 400000, 'Max memory usage: %dkb' % usage)
-            if self.swapctl == 'true':
+        # loop to check if cput, mem, vmem are expected values
+        cput_usage = 0.0
+        mem_usage = 0
+        vmem_usage = 0
+        for count in range(3):
+            # Faster systems might have expected usage after 8 seconds
+            time.sleep(8)
+            if self.paths['cpuacct'] and cput_usage <= 1.0:
                 lines = self.moms_list[0].log_match(
-                    '%s;update_job_usage: Memory usage: vmem=' % jid,
-                    allmatch=True, max_attempts=5, starttime=begin)
-                usage = 0
+                    '%s;update_job_usage: CPU usage:' %
+                    jid, allmatch=True, starttime=begin)
                 for line in lines:
-                    match = re.search(r'vmem=(\d+)kb', line[1])
+                    match = re.search(r'CPU usage: ([0-9.]+) secs', line[1])
                     if not match:
                         continue
-                    usage = int(match.groups()[0])
-                    if usage > 400000:
+                    cput_usage = float(match.groups()[0])
+                    if cput_usage > 1.0:
                         break
-                self.assertGreater(usage, 400000)
+            if self.paths['memory'] and mem_usage <= 400000:
+                lines = self.moms_list[0].log_match(
+                    '%s;update_job_usage: Memory usage: mem=' % jid,
+                    allmatch=True, starttime=begin)
+                for line in lines:
+                    match = re.search(r'mem=(\d+)kb', line[1])
+                    if not match:
+                        continue
+                    mem_usage = int(match.groups()[0])
+                    if mem_usage > 400000:
+                        break
+                if self.swapctl == 'true' and vmem_usage <= 400000:
+                    lines = self.moms_list[0].log_match(
+                        '%s;update_job_usage: Memory usage: vmem=' % jid,
+                        allmatch=True, starttime=begin)
+                    for line in lines:
+                        match = re.search(r'vmem=(\d+)kb', line[1])
+                        if not match:
+                            continue
+                        vmem_usage = int(match.groups()[0])
+                        if vmem_usage > 400000:
+                            break
+        self.assertGreater(cput_usage, 1.0)
```

Comment:
You should check cput, mem_usage, and vmem_usage at the end of the loop. If they are all above the expected values, then break from the loop. Otherwise, one may be correct but the others are not.

Reply:
We might not get the expected values within the loop, a reasonable time frame within which we can expect the correct values.  So I check the cput, mem_usage, and vmem_usage at the end of the loop.

Reply:
Same WRT removing the conditional for the break. Also, you should check all three at the end of the loop. If they are all above the expected values then you can break from the 8 second sleep loop. No need to iterate again if all the values are where they need to be.

Reply:
I check that the cput, mem, vmem values match what is expected within each loop. If within a loop the expected value is reached then that parameter is not checked in the next loop. If all the parameters have reached expected values then the mom logs are not reread in the following loop. If at least one parameter has not reached the expected value after all the loops are exhausted then we get an error.

Reply:
Thanks for the explanation. If all three values are above the threshold we're looking for upon the first iteration, what stops the loop from executing again and sleeping for another 8 seconds?

Reply:
Per your suggestion I have added a check for all three at the end of each loop and break if appropriate.  This will stop looping again and sleeping for 8 seconds.

Reply:
What happens when cput_usage > 1.0 and mem_usage > 400000 and self.swapctl == 'true'? The loop will break whether or not vmem_usage > 400000. Also, I find it very strange that self.swapctl is a string and not a boolean. I think is_memsw_enabled() needs updating.

Reply:
I have corrected the value checks at the end of each loop.  Changing is_memsw_enabled() to return a boolean (e.g. True) will affect the settings in the config files which needs 'true', for example:
        "memsw":
        {
            "enabled"         :  true